### DISCUSSION TITLE
Do you consider Rust ?

### DISCUSSION THREAD

**Discussion by monkeycc** [ultralytics/ultralytics#3861] [General]
Recently studying Rust

Discovered that deployment is quite convenient

I'm not sure if it can be applied to Yolov8

Want to use Yolov8 in Rust

----------

High performance

Low resource occupancy

Can be deployed on the desktop WebAssembly embedded development board server

The server is suitable for large-scale concurrent inference

Examples of frameworks for deep learning, etc

https://github.com/LaurentMazare/tch-rs
https://github.com/burn-rs/burn

https://github.com/pykeio/ort
https://github.com/microsoft/onnxruntime/tree/main/rust

https://github.com/AndreyGermanov/yolov8_onnx_rust
https://github.com/igor-yusupov/rusty-yolo
https://github.com/bencevans/rust-opencv-yolov5
https://github.com/masc-it/yolov5-api-rust

WebAssembly Demo

https://burn-rs.github.io/demo

----------

I think it can be achieved

But the level is limited

**Comment by glenn-jocher** [comment]
@monkeycc hello! It's great to hear about your interest in using Rust for deploying YOLOv8 models. Rust's performance and low resource footprint can indeed be beneficial for various deployment scenarios.

While YOLOv8 itself is not written in Rust, you can certainly use the model within a Rust environment. One common approach is to export the YOLOv8 model to ONNX format using the Export mode documented in our Ultralytics Docs. Once you have the ONNX model, you can leverage Rust libraries like `onnxruntime` for inference.

For deployment on platforms like WebAssembly, you would typically follow a similar process: export the model and then use a compatible WebAssembly runtime to execute it in your desired environment.

Although we don't provide native Rust support or examples, the community has been active in creating Rust bindings and APIs for deep learning models, as you've found. You can use these resources as a starting point and adapt them for YOLOv8.

If you encounter any specific issues while working with YOLOv8 in your Rust application, feel free to reach out, and we'll do our best to assist you. Good luck with your project! 🚀
